File: ng-frontend/src/app/modules/tasks/model/task.ts
import {UserService} from "../../../services/user.service";
import {User} from "../../user-account/model/user";
import {Priority} from "./priority";
import {Project} from "../../projects/model/project";
import {Status} from "./status";

export class Task {

  id: number;
  taskCode: number;
  projectCode: Project;
  description: string;
  priority: Priority;
  reporter: User;
  assignedUser: User;
  dueDate: number;
  estimation: number;
  updatedDate: number;
  createdDate: number;
  status: Status;


  static cloneTask(task: Task): Task {
    const clonedTask: Task = new Task();

    clonedTask.assignedUser = task.assignedUser;
    clonedTask.createdDate = task.createdDate;
    clonedTask.description = task.description;
    clonedTask.dueDate = task.dueDate;
    clonedTask.estimation = task.estimation;
    clonedTask.priority = task.priority;
    clonedTask.reporter = task.reporter;
    clonedTask.taskCode = task.taskCode;
    clonedTask.projectCode = task.projectCode;
    clonedTask.updatedDate  = task.updatedDate;
    clonedTask.id = task.id;
    clonedTask.status = task.status;
    return clonedTask;
  }
}

File: fapi/src/main/java/com/netcracker/projectmanager/models/Task.java
package com.netcracker.projectmanager.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.Objects;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Task {


    private Integer id;

    private Integer ticketCode;

     private String description;

    private Priority priority;

    private Project projectCode;

    private Status status;

    private Long dueDate;

    private Long estimation;

    private Long createdDate;

    private Long updatedDate;


    private User reporter;


    private User assignedUser;

    public Task() {
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getTaskCode() {
        return ticketCode;
    }

    public void setTaskCode(Integer taskCode) {
        this.ticketCode = taskCode;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Integer getTicketCode() {
        return ticketCode;
    }

    public void setTicketCode(Integer ticketCode) {
        this.ticketCode = ticketCode;
    }

    public Priority getPriority() {
        return priority;
    }

    public void setPriority(Priority priority) {
        this.priority = priority;
    }

    public Project getProjectCode() {
        return projectCode;
    }

    public void setProjectCode(Project projectCode) {
        this.projectCode = projectCode;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public User getReporter() {
        return reporter;
    }

    public void setReporter(User reporter) {
        this.reporter = reporter;
    }

    public User getAssignedUser() {
        return assignedUser;
    }

    public void setAssignedUser(User assignedUser) {
        this.assignedUser = assignedUser;
    }

    public Long getDueDate() {
        return dueDate;
    }

    public void setDueDate(Long dueDate) {
        this.dueDate = dueDate;
    }

    public Long getEstimation() {
        return estimation;
    }

    public void setEstimation(Long estimation) {
        this.estimation = estimation;
    }

    public Long getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Long createdDate) {
        this.createdDate = createdDate;
    }

    public Long getUpdatedDate() {
        return updatedDate;
    }

    public void setUpdatedDate(Long updatedDate) {
        this.updatedDate = updatedDate;
    }



    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Task task = (Task) o;
        return id.equals(task.id) &&
                ticketCode.equals(task.ticketCode) &&
                projectCode.equals(task.projectCode);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, ticketCode, projectCode);
    }
}
